common/auth.js

```diff
@@ -103,6 +105,18 @@ class Auth {
     Auth.#authConfig = options.authConfig;
     Auth.#caTrustCerts = ArkimeUtil.certificateFileToArray(options.caTrustFile);
 
+    if (Auth.#app && options.trustProxy !== undefined) {
+      if (options.trustProxy === true || options.trustProxy === 'true') {
+        Auth.#app.set('trust proxy', true);
+      } else if (options.trustProxy === false || options.trustProxy === 'false') {
+        Auth.#app.set('trust proxy', false);
+      } else if (!isNaN(options.trustProxy)) {
+        Auth.#app.set('trust proxy', parseInt(options.trustProxy));
+      } else {
+        Auth.#app.set('trust proxy', options.trustProxy);
+      }
+    }
```

Comment:
There are two issues here.

First:


```suggestion
    if (Auth.#app && Auth.#authConfig.trustProxy !== undefined) {
      if (Auth.#authConfig.trustProxy === true || Auth.#authConfig.trustProxy === 'true') {
        Auth.#app.set('trust proxy', true);
      } else if (Auth.#authConfig.trustProxy === false || Auth.#authConfig.trustProxy === 'false') {
        Auth.#app.set('trust proxy', false);
      } else if (!isNaN(Auth.#authConfig.trustProxy)) {
        Auth.#app.set('trust proxy', parseInt(Auth.#authConfig.trustProxy));
      } else {
        Auth.#app.set('trust proxy', Auth.#authConfig.trustProxy);
      }
    }
```


Second:
`Auth.#app` is unset because Config calls `Auth.initalize()` very early in viewer's setup:
https://github.com/arkime/arkime/blob/trust-proxy/viewer/viewer.js#L25
https://github.com/arkime/arkime/blob/trust-proxy/viewer/config.js#L505

which is well before the call to `Auth.app(app);`:
https://github.com/arkime/arkime/blob/trust-proxy/viewer/viewer.js#L262

(I don't have a great way to test Arkime built from source right now, so I applied this PR's patch onto v4.2.0 instead. It's possible my test missed something)